server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -45,6 +51,25 @@ public class HealthPeriodicLogger implements ClusterStateListener, Closeable, Sc
     public static final String HEALTH_FIELD_PREFIX = "elasticsearch.health";
     public static final String MESSAGE_FIELD = "message";
 
+    // Writers for logs or messages
+    // default visibility for testing purposes
+    BiConsumer<LongGaugeMetric, Long> metricWriter = LongGaugeMetric::set;
+    Consumer<ESLogMessage> logWriter = logger::info;
+
+    /**
+     * Valid modes of output for this logger
+     */
+    public enum OutputMode {
+        LOGS,
+        METRICS,
+        BOTH;
```

Comment:
I think I'd prefer to call this "all", so that in the future if we ever add a third option then "both" doesn't mean three things.

Reply:
Actually, thinking about this more, if we _ever_ envision a future where we may have 3+ options, then it might make sense to make this an array setting so a user can do `output_mode: logs,metrics` (and then potentially something like `output_mode: metrics,carrier_pigeon` down the road). What do you think?